src/include/miopen/backend_api.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <miopen/miopen.h>
+#include <vector>
+
+namespace miopen {
+
+namespace backend_api {
+
+class BackendDescriptor
+{
+public:
+    BackendDescriptor() : m_IsFinalized(false) {}
+    virtual ~BackendDescriptor()                               = default;
```

Comment:
define the destructor in `backend_api.cpp` as that allows the compiler to `anchor` the virtual destructor.

Reply:
(a short explanation)
https://llvm.org/docs/CodingStandards.html#provide-a-virtual-method-anchor-for-classes-in-headers